index.bs

```diff
@@ -3481,9 +3391,9 @@ and at least one of the following three conditions is true:
     </table>
 
     <ol type="a">
-        1.  The two identified [=interfaces=] are
-            not the same, it is not possible for a single [=platform object=]
-            to implement both [=interfaces=],
+        1.  The two identified interface-like types are
+            not the same, no single [=platform object=] implements both
+            interface-like types,
             and it is not the case that both are [=callback interfaces=].
```

Comment:
I suspect we can make this easier to read by splitting callback interfaces out of interface-like, and maybe moving them into a dictionary-like category. I didn't do that yet in order to make this change easier to review, but I could if y'all want.